src/transformers/generation_utils.py

```diff
@@ -618,6 +628,8 @@ def generate(
                 eos_token_id=eos_token_id,
                 **model_kwargs,
             )
+        else:
```

Comment:
I don't think it's possible to arrive here if `num_beams` is set correctly

Reply:
Agreed - once we have the num_beams check this could never happen. (I mentioned that in the PR note!) 

It just seems like good code style to avoid having the possibility of falling off the end if changes are ever made to the code above?

Reply:
Ok for me. Can we change the error message though, *e.g.*:

"No generation method was called. Make sure to correctly set the parameters `num_beams` and `do_sample` to call one of `greedy_search`, `sample`, `beam_seach`, or `beam_sample`."